drakrun/drakrun/main.py

```diff
@@ -46,6 +48,28 @@ def start_tcpdump_collector(instance_id: str, outdir: str) -> subprocess.Popen:
     ])
 
 
+def start_dnsmasq(vm_id: int) -> Optional[subprocess.Popen]:
+    try:
+        subprocess.check_output("dnsmasq --version", shell=True)
+    except subprocess.CalledProcessError:
+        logging.warning("Seems like dnsmasq is not working/not installed on your system."
+                        "Guest networking may not be fully functional.")
+        return
+
+    return subprocess.Popen([
+        "dnsmasq",
+        "--no-daemon",
+        f"--interface=drak{vm_id}",
+        "--port=0",
+        "--no-hosts",
+        "--no-resolv",
+        "--no-poll",
+        "--leasefile-ro",
+        f"--dhcp-range=10.13.{vm_id}.100,10.13.{vm_id}.200,255.255.255.0,12h",
+        "--dhcp-option=option:dns-server,8.8.8.8"
```

Comment:
I would recommend running a local DNS server so you can log the queries. I've used dnschef in the past and it was quite straight forward.

Reply:
I think we already capture DNS queries in a tcpdump `pcap`, however the DNS could be actually configurable and there could be an optional step of adding dnschef :+1:

Reply:
@tklengyel Do you have some experience with network issues when doing a VM cloning? Were you just leaving the same MAC for each VM or doing some other stuff to fix it?

Reply:
MAC address collision is a problem. TCP connections get reset if there are multiple clones active and they don't expect the TCP stream. At one point I wrote my own NAT engine that routed based on vlans instead of MAC/IP using Honeybrid. Otherwise you have to reconfigure each clone to assign new MACs after they are running.

Reply:
Vlans also solve the problem. You only have to inject a command to set unique IP for each clone, then on the vlan there is no MAC collision.